video-app/src/app/library/add/add.component.ts

```diff
@@ -34,25 +41,49 @@ export class AddComponent {
   constructor(private readonly router: Router
     ,         private readonly youtubeService: YoutubeService
     ,         private readonly vimeoService: VimeoService
-    ,         private readonly streamingPlatform: StreamingPlatformService) { }
+    ,         private readonly streamingPlatform: StreamingPlatformService
+    ,         private readonly store: Store<{ videos: VideoState }>) {
+      this.video$ = store.pipe(select('videos'));
+     }
+
+    public ngOnInit(): void {
+      this.videoSubscription = this.video$
+        .pipe(
+          map(x => {
+            this.showErrorMessage = x.ShowErrorMessage;
+
+            // ToDo: how to redirect to home page
+            // if (!this.showErrorMessage) {
+            //   this.router.navigate(['/home']);
+            // }
+          })
+        )
+        .subscribe();
+    }
+
+    public ngOnDestroy(): void {
+      if (this.videoSubscription) {
+        this.videoSubscription.unsubscribe();
```

Comment:
https://github.com/w11k/ngx-componentdestroyed
możesz to wpiąć, przyjemniej się wtedy robi unsubscibe